tests/integration/producer/test_transactions.py

```diff
@@ -0,0 +1,189 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limit
+#
+from uuid import uuid1
+
+from confluent_kafka import Consumer, KafkaError
+
+
+def my_error_cb(err):
+    """ Reports global/generic errors to aid in troubleshooting test failures. """
+    print(err)
+
+
+def delivery(err, msg):
+    """ Reports failed message delivery to aid in troubleshooting test failures. """
+    if err:
+        print('Message delivery failed (%s [%s]): %s' %
+              (msg.topic(), str(msg.partition()), err))
+        return
+
+
+def test_commit_transaction(kafka_cluster):
+    output_topic = kafka_cluster.create_topic("output_topic" + str(uuid1()))
+
+    producer = kafka_cluster.producer({
+        'transactional.id': 'example_transactional_id',
+        'error_cb': my_error_cb,
+
+    })
+
+    producer.init_transactions()
+    transactional_produce(producer, output_topic, 100)
+    producer.commit_transaction()
+
+    assert consume_committed(kafka_cluster.client_conf(), output_topic) == 100
+
+
+def test_abort_transaction(kafka_cluster):
+    output_topic = kafka_cluster.create_topic("output_topic" + str(uuid1()))
+
+    producer = kafka_cluster.producer({
+        'transactional.id': 'example_transactional_id',
+        'error_cb': my_error_cb,
+    })
+
+    producer.init_transactions()
+    transactional_produce(producer, output_topic, 100)
+    producer.abort_transaction()
+
+    assert consume_committed(kafka_cluster.client_conf(), output_topic) == 0
+
+
+def test_abort_retry_commit_transaction(kafka_cluster):
+    output_topic = kafka_cluster.create_topic("output_topic" + str(uuid1()))
+
+    producer = kafka_cluster.producer({
+        'transactional.id': 'example_transactional_id',
+        'error_cb': my_error_cb,
+    })
+
+    producer.init_transactions()
+    transactional_produce(producer, output_topic, 100)
+    producer.abort_transaction()
+
+    transactional_produce(producer, output_topic, 25)
+    producer.commit_transaction()
+
+    assert consume_committed(kafka_cluster.client_conf(), output_topic) == 25
+
+
+def test_send_offsets_committed_transaction(kafka_cluster):
+    input_topic = kafka_cluster.create_topic("input_topic" + str(uuid1()))
+    output_topic = kafka_cluster.create_topic("output_topic" + str(uuid1()))
+
+    producer = kafka_cluster.producer({
+        'client.id': 'producer1',
+        'transactional.id': 'example_transactional_id',
+        'error_cb': my_error_cb,
+    })
+
+    consumer_conf = {
+        'group.id': str(uuid1()),
+        'auto.offset.reset': 'earliest',
+        'isolation.level': 'read_committed',
```

Comment:
that's the default, isnt it?

Reply:
yes, although this isn't differs from java and makes following the blog on EOS a bit odd. Cleaning up to highlight the better default in librdkafka.